**Functions in Python**
* def function_name(parameters):
  * #statementblock

In [1]:
def welcome():
  print('Welcome to python functions')

In [2]:
welcome()

Welcome to python functions


In [3]:
for i in range(5):
  welcome()

Welcome to python functions
Welcome to python functions
Welcome to python functions
Welcome to python functions
Welcome to python functions


In [4]:
#functions with parameter
def display(username):
  print('Welcome to Python functions Mr.\\Ms.',username)

In [5]:
display('sam')

Welcome to Python functions Mr.\Ms. sam


In [7]:
username=input("enter username:")
display(username)

enter username:nilesh
Welcome to Python functions Mr.\Ms. nilesh


In [8]:
def addition(num1,num2):
  print('First number is:',num1)
  print('Second number is:',num2)
  result=num1+num2
  print('Addition is:',result)

In [10]:
first_num=float(input('Enter first number:'))
second_num=float(input('Enter second number:'))
addition(first_num,second_num)

Enter first number:12.5
Enter second number:13.75
First number is: 12.5
Second number is: 13.75
Addition is: 26.25


In [12]:
#  return type
def multiplication(num1,num2):
  result=num1*num2
  return result

In [15]:
res= multiplication(12,15)
print('Result: ',res)

Result:  180


In [18]:
fnum=float(input('Enter first number:'))
snum=float(input('Enter second number:'))
res=multiplication(fnum,snum)
print(f'After multiplying {fnum} and {snum} Result is :  ',res)

Enter first number:125
Enter second number:2.45
After multiplying 125.0 and 2.45 Result is :   306.25


### Write a function to calculate the cube of number entered by user.
* Take the number from user and display following details
* for example Number: 5 Cube : 125
* for example Number: 2 Cube : 8

In [19]:
def cube(number):
  print(f'Number: {number} Cube: {number**3}')

In [20]:
cube(5)

Number: 5 Cube: 125


In [21]:
cube(2)

Number: 2 Cube: 8


In [22]:
cube(9)

Number: 9 Cube: 729


1. Poistional
2. Default
3. Variable Length
4. keyword arguments



In [23]:
def reg(name,age,gender):
  print('Registration success details as follows')
  print('Name:',name)
  print('Age:',age)
  print('Gender:',gender)

In [24]:
reg('sam',25,'male')

Registration success details as follows
Name: sam
Age: 25
Gender: male


In [26]:
reg(age=25, gender='female',name='sam')

Registration success details as follows
Name: sam
Age: 25
Gender: female


In [28]:
#default parameter
def reg(name,age,gender='male'):
  print('Registeration Success')
  print('Name:',name)
  print('Age:',age)
  print('Gender:',gender)

In [29]:
reg('Sam',25)

Registeration Success
Name: Sam
Age: 25
Gender: male


In [30]:
reg('Ria',23,'female')

Registeration Success
Name: Ria
Age: 23
Gender: female


In [32]:
reg('neel',45)

Registeration Success
Name: neel
Age: 45
Gender: male


In [33]:
#we can pass multiple default parameters in parameter list of function
#but default parameter must be last in parameter list
def add_num(num1,num2=10,num3=5, num4=20):
  result=num1+num2+num3+num4
  print(f'Result after adding {num1},{num2},{num3},{num4} is = \t {result}')

In [34]:
add_num(10)

Result after adding 10,10,5,20 is = 	 45


In [35]:
add_num(10,20)

Result after adding 10,20,5,20 is = 	 55


In [36]:
add_num(45,12,13,11)

Result after adding 45,12,13,11 is = 	 81


In [39]:
#variable length parameters or arguments
def add_num(*num):
  total=0
  print('Total Number passed by you are:',len(num))
  for i in num:
        total=total+i
  print(f'Result after adding {num} is = \t {total}')


In [40]:
add_num(12,11)

Total Number passed by you are: 2
Result after adding (12, 11) is = 	 23


In [42]:
add_num(10,3,5,6,8)

Total Number passed by you are: 5
Result after adding (10, 3, 5, 6, 8) is = 	 32


In [43]:
add_num(45,11,34)

Total Number passed by you are: 3
Result after adding (45, 11, 34) is = 	 90


In [44]:
# 4. keyword arguments **kwargs
def display(**kwargs):
  print(kwargs)


In [45]:
display(name='sam',age=25,gender='male')

{'name': 'sam', 'age': 25, 'gender': 'male'}


In [46]:
display(id=12)

{'id': 12}


In [47]:
display(name='sam',country='india')

{'name': 'sam', 'country': 'india'}


In [50]:
def hotel_room_booking(customer_name,room_no,**others):
 print('Booking Details as follows')
 print('Customer Name:',customer_name)
 print('Room No:',room_no)
 if others:
   print('Other Details')
   for key,value in others.items():
     print(key,':',value)


In [51]:
hotel_room_booking('sam',123)

Booking Details as follows
Customer Name: sam
Room No: 123


In [52]:
hotel_room_booking('arsh',12,nights=5,breakfast='yes',lunch='yes')

Booking Details as follows
Customer Name: sam
Room No: 12
Other Details
nights : 5
breakfast : yes
lunch : yes
